### PR TITLE
Use browserify packages directly

### PR DESCRIPTION
**Pull request by dcousens**
This PR is not-dissimilar to using `inherits` over `util.inherits`.

It also exposed a [bug](https://github.com/crypto-browserify/pbkdf2/issues/11) (in Node?), and is therefore blocked by it for now.

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -51,13 +49,11 @@ describe('BIP39', function() {
...
-      it('works for tests vector ' + i, sinon.test(function() {
-        this.mock(crypto).expects('randomBytes')
-          .exactly(1)
-          .onCall(0).returns(new Buffer(v[0], 'hex'))
+      it('works for tests vector ' + i, function() {
+        function rng() { return new Buffer(v[0], 'hex') }
 
-        assert.equal(BIP39.generateMnemonic(), v[1])
-      }))
+        assert.equal(BIP39.generateMnemonic(undefined, rng), v[1])
```
Comment (weilu) [inline test/index.js]: This change left the default randomBytes code path uncovered.

Comment (dcousens) [inline test/index.js]: No it doesn't?  See https://github.com/weilu/bip39/blob/packdirect/test/index.js#L59

**Comment by dcousens** [comment]
Good to merge,  `pbkdf2` encoding was fixed in https://github.com/crypto-browserify/pbkdf2/pull/12#discussion_r27272807

The change pulled `pkbdf2` in line with node, which uses `binary` encoding for strings, totally broken IMHO.  Anyway, to avoid this, I explicitly create the `Buffer`s in d99ec86.

**Comment by dcousens** [comment]
Please publish ASAP :)

File: test/index.js (hunk 1 above)
Comment (weilu) [inline test/index.js]: The test you pointed me to is only testing the resulting word length. It has no assertion on which RNG we use as the default whatsoever. I think the default RNG is important enough for us to put some assertion around it. The original test at least says "we use the RNG returned by the `randombytes` module and we expect the module to do the right thing, whatever the right thing is defined in that library", in other words, the responsibility of using a cryptographically secure RNG is delegated to `randombytes`.

Comment (weilu) [inline test/index.js]: @dcousens I need the test back before I can merge, tag & release. If you don't have a chance to do it I can do it over the weekend.

**Comment by rubensayshi** [comment]
ping @weilu this is good to go right?

**Comment by dcousens** [comment]
I think a test case to verify the correct RNG is being called needs to be
added yet.  Then it's good to merge.  Will do tomorrow.

**Comment by dcousens** [comment]
@weilu added test for randombytes,  ready to merge, tag and publish :)

**Comment by dcousens** [comment]
Cheers :)

**Comment by weilu** [comment]
Will tag & release when I get my hands on a real keyboard

**Comment by dcousens** [comment]
Sweet, so I assume that will be in about 4 hours? :)

**Comment by weilu** [comment]
More like 8 but it's done :)

**Pull request merged**